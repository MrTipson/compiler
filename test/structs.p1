void putint(int val);
int write(int fd, char* cbuf, int count);

struct Point {
	int x;
	bool isImportant;
	int y;
};

int main(int argc, char** argv) {
	Point p;
	p.x = 5;
	p.isImportant = false;
	p.y = 10;
	printPoint(&p);
	Point r;
	r.x = -1;
	r.isImportant = true;
	r.y = 24;
	printPoint(&r);
}

void printPoint(Point* point) {
	write(1,"( ",2);
	putint(point->x);
	write(1,", ",2);
	putint(point->y);
	write(1," ) is ", 6);
	if(!point->isImportant) write(1, "not ", 4);
	write(1,"important.\n",11);
}