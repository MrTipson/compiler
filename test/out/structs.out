( 5, 10 ) is not important.abc
( -1, 24 ) is important.cba
